stix_shifter_modules/nozomi/stix_transmission/connector.py

```diff
@@ -0,0 +1,217 @@
+
+        except Exception as ex:
+            return_obj = self.exception_response(None, str(ex))
+        return return_obj
+
+    async def ping_connection(self):
+        """
+        Ping the endpoint
+        :return: return_object, dict
+        """
+        return_obj = {}
+        try:
+            token = await self.get_token()
+            response = await self.api_client.ping_data_source(token)
+            response_code = response.code
+            response_dict = json.loads(response.read().decode('utf-8'))
+            if response_code == 200:
+                return_obj['success'] = True
+            else:
+                return_obj = self.exception_response(response_code, response_dict.get('errorSummary', ''))
+        except Exception as ex:
+            return_obj = self.exception_response(None, str(ex))
+        return return_obj
+
+    async def get_token(self):
+        """ Generate new token"""
+        response = await self.api_client.generate_token()
+        response_code = response.code
+        response_txt = response.read().decode('utf-8')
+        response_json = json.loads(response_txt)
+        if response_code == 200:
+            response_header = response.headers
+            if 'Authorization' in response_header:
+                token = response_header.get('Authorization')
+            return token
+        raise Exception(response_json['errors'])
+
+    def exception_response(self, code=None, response_txt=''):
+        """
+        create the exception response
+        :param code, int
+        :param response_txt, str
+        :return: return_obj, dict
+        """
+        return_obj = {}
+        message = None
+
+        if 'Query is not valid' in response_txt:
+            code = 400
+            message = 'Query is not valid'
+
+        if code == 401 and not response_txt:
+            message = 'Authentication failed'
+
+        if not code:
+            if "'key_name': ['is invalid']" in response_txt:
+                code = 401
+                message = "Invalid Authentication: " + response_txt
+            elif "timeout_error" in response_txt:
+                code = 408
+            elif 'Invalid URL' in response_txt:
+                code = 408
+                message = "Invalid Host: " + response_txt
+
+        if not message:
+            message = str(response_txt)
+
+        response_dict = {'code': code, 'message': message}
+        self.logger.error('%s error while fetching results: %s', self.connector, message)
+        ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+        return return_obj
+
+    @staticmethod
+    def get_results_data(response):
+        """
+         Preprocessing the response.
+        :param response: list
+        :return response: list
+        """
+        for record in response:
+            # threat_name is None or '' means finding type is alert"
+            if record.get('threat_name') is None:
```

Comment:
You are only checking for None here and not '' (empty). Can you clarify in the comment or correct the statement to check for ''?

Reply:
if threat_name is None or '' value should be alert. Added '' value in the check.